### PR TITLE
fix: resolve preset-modules

### PR DESCRIPTION
**Pull request by merceyz**
https://github.com/zeit/next.js/pull/9489 started using `@babel/preset-modules` without using `require`, this is unsafe and doesn't work with PnP

`@babel/preset-modules` doesn't support pnp as it also doesn't use require, but there is a PR for this upstream, see https://github.com/babel/preset-modules/pull/6

### CODE REVIEW

File: packages/next/build/webpack/plugins/next-esm-plugin.ts [hunk 1]
```diff
@@ -169,7 +169,7 @@ export default class NextEsmPlugin implements Plugin {
           }
 
           if (IS_PRESET_ENV.test(name)) {
-            presets.push(['@babel/preset-modules', { loose: true }])
+            presets.push([require('@babel/preset-modules'), { loose: true }])
```
Comment (Timer) [inline packages/next/build/webpack/plugins/next-esm-plugin.ts]: Please make this `require.resolve`, which I think we use everywhere else:
```suggestion
            presets.push([require.resolve('@babel/preset-modules'), { loose: true }])
```

**Review by Timer — CHANGES_REQUESTED** [review]

File: packages/next/build/webpack/plugins/next-esm-plugin.ts (hunk 1 above)
Comment (merceyz) [reply packages/next/build/webpack/plugins/next-esm-plugin.ts]: I'll change it if you want, but https://github.com/zeit/next.js/blob/dee28884042a7d5485166c961bd4a6d514260b5f/packages/next/build/babel/preset.ts is using `require`

Comment (Timer) [reply packages/next/build/webpack/plugins/next-esm-plugin.ts]: Interesting -- it shouldn't be, please change this! 🙏

**Review by Timer — APPROVED** [review]
Thanks!

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| buildDuration | 14.5s | 14.5s | ⚠️ +19ms |
| nodeModulesSize | 48.6 MB | 48.6 MB | ⚠️ +34 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| main-HASH.js | 18.1 kB | 18.1 kB | ✓ |
| main-HASH.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-HASH.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..4086094cc.js | 19.8 kB | 19.8 kB | ✓ |
| 16b1a7225520..94cc.js gzip | 7.37 kB | 7.37 kB | ✓ |
| 4952ddcd88e7..49d7903c2.js | 9.6 kB | 9.6 kB | ✓ |
| 4952ddcd88e7..03c2.js gzip | 3.76 kB | 3.76 kB | ✓ |
| commons.HASH.js | 10.4 kB | 10.4 kB | ✓ |
| commons.HASH.js gzip | 3.91 kB | 3.91 kB | ✓ |
| de003c3a9d30..64b97c6d1.js | 21.1 kB | 21.1 kB | ✓ |
| de003c3a9d30..c6d1.js gzip | 8.15 kB | 8.15 kB | ✓ |
| framework.HASH.js | 125 kB | 125 kB | ✓ |
| framework.HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 206 kB | 206 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| main-HASH.module.js | 14.1 kB | 14.1 kB | ✓ |
| main-HASH.module.js gzip | 5.33 kB | 5.33 kB | ✓ |
| webpack-HASH.module.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..95.module.js | 21.1 kB | 21.1 kB | ✓ |
| 16b1a7225520..dule.js gzip | 8.14 kB | 8.14 kB | ✓ |
| 4952ddcd88e7..cc.module.js | 11.9 kB | 11.9 kB | ✓ |
| 4952ddcd88e7..dule.js gzip | 4.41 kB | 4.41 kB | ✓ |
| de003c3a9d30..f3.module.js | 14.9 kB | 14.9 kB | ✓ |
| de003c3a9d30..dule.js gzip | 6.11 kB | 6.11 kB | ✓ |
| framework.HASH.module.js | 125 kB | 125 kB | ✓ |
| framework.HA..dule.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 189 kB | 189 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| polyfills-HASH.js | 15.3 kB | 15.3 kB | ✓ |
| polyfills-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| Overall change | 15.3 kB | 15.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _app.js | 2.92 kB | 2.92 kB | ✓ |
| _app.js gzip | 1.32 kB | 1.32 kB | ✓ |
| _error.js | 13.2 kB | 13.2 kB | ✓ |
| _error.js gzip | 5.1 kB | 5.1 kB | ✓ |
| hooks.js | 1.92 kB | 1.92 kB | ✓ |
| hooks.js gzip | 941 B | 941 B | ✓ |
| index.js | 318 B | 318 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js | 6.76 kB | 6.76 kB | ✓ |
| link.js gzip | 2.88 kB | 2.88 kB | ✓ |
| routerDirect.js | 413 B | 413 B | ✓ |
| routerDirect.js gzip | 285 B | 285 B | ✓ |
| withRouter.js | 423 B | 423 B | ✓ |
| withRouter.js gzip | 284 B | 284 B | ✓ |
| Overall change | 25.9 kB | 25.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _app.module.js | 1.53 kB | 1.53 kB | ✓ |
| _app.module.js gzip | 745 B | 745 B | ✓ |
| _error.module.js | 10.3 kB | 10.3 kB | ✓ |
| _error.module.js gzip | 4.24 kB | 4.24 kB | ✓ |
| hooks.module.js | 651 B | 651 B | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js | 276 B | 276 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js | 5.44 kB | 5.44 kB | ✓ |
| link.module.js gzip | 2.45 kB | 2.45 kB | ✓ |
| routerDirect.module.js | 386 B | 386 B | ✓ |
| routerDirect..dule.js gzip | 275 B | 275 B | ✓ |
| withRouter.module.js | 396 B | 396 B | ✓ |
| withRouter.m..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 19 kB | 19 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _buildManifest.js | 244 B | 244 B | ✓ |
| _buildManifest.js gzip | 200 B | 200 B | ✓ |
| _buildManifest.module.js | 251 B | 251 B | ✓ |
| _buildManife..dule.js gzip | 210 B | 210 B | ✓ |
| Overall change | 495 B | 495 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| index.html | 4.06 kB | 4.06 kB | ✓ |
| index.html gzip | 1.03 kB | 1.03 kB | ✓ |
| link.html | 4.58 kB | 4.58 kB | ✓ |
| link.html gzip | 1.11 kB | 1.11 kB | ✓ |
| withRouter.html | 4.59 kB | 4.59 kB | ✓ |
| withRouter.html gzip | 1.1 kB | 1.1 kB | ✓ |
| Overall change | 13.2 kB | 13.2 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| buildDuration | 15.1s | 15.2s | ⚠️ +130ms |
| nodeModulesSize | 48.6 MB | 48.6 MB | ⚠️ +34 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| main-HASH.js | 18.1 kB | 18.1 kB | ✓ |
| main-HASH.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-HASH.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..4086094cc.js | 19.8 kB | 19.8 kB | ✓ |
| 16b1a7225520..94cc.js gzip | 7.37 kB | 7.37 kB | ✓ |
| 4952ddcd88e7..49d7903c2.js | 9.6 kB | 9.6 kB | ✓ |
| 4952ddcd88e7..03c2.js gzip | 3.76 kB | 3.76 kB | ✓ |
| commons.HASH.js | 10.4 kB | 10.4 kB | ✓ |
| commons.HASH.js gzip | 3.91 kB | 3.91 kB | ✓ |
| de003c3a9d30..64b97c6d1.js | 21.1 kB | 21.1 kB | ✓ |
| de003c3a9d30..c6d1.js gzip | 8.15 kB | 8.15 kB | ✓ |
| framework.HASH.js | 125 kB | 125 kB | ✓ |
| framework.HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 206 kB | 206 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| main-HASH.module.js | 14.1 kB | 14.1 kB | ✓ |
| main-HASH.module.js gzip | 5.33 kB | 5.33 kB | ✓ |
| webpack-HASH.module.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..95.module.js | 21.1 kB | 21.1 kB | ✓ |
| 16b1a7225520..dule.js gzip | 8.14 kB | 8.14 kB | ✓ |
| 4952ddcd88e7..cc.module.js | 11.9 kB | 11.9 kB | ✓ |
| 4952ddcd88e7..dule.js gzip | 4.41 kB | 4.41 kB | ✓ |
| de003c3a9d30..f3.module.js | 14.9 kB | 14.9 kB | ✓ |
| de003c3a9d30..dule.js gzip | 6.11 kB | 6.11 kB | ✓ |
| framework.HASH.module.js | 125 kB | 125 kB | ✓ |
| framework.HA..dule.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 189 kB | 189 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| polyfills-HASH.js | 15.3 kB | 15.3 kB | ✓ |
| polyfills-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| Overall change | 15.3 kB | 15.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _app.js | 2.92 kB | 2.92 kB | ✓ |
| _app.js gzip | 1.32 kB | 1.32 kB | ✓ |
| _error.js | 13.2 kB | 13.2 kB | ✓ |
| _error.js gzip | 5.1 kB | 5.1 kB | ✓ |
| hooks.js | 1.92 kB | 1.92 kB | ✓ |
| hooks.js gzip | 941 B | 941 B | ✓ |
| index.js | 318 B | 318 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js | 6.76 kB | 6.76 kB | ✓ |
| link.js gzip | 2.88 kB | 2.88 kB | ✓ |
| routerDirect.js | 413 B | 413 B | ✓ |
| routerDirect.js gzip | 285 B | 285 B | ✓ |
| withRouter.js | 423 B | 423 B | ✓ |
| withRouter.js gzip | 284 B | 284 B | ✓ |
| Overall change | 25.9 kB | 25.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _app.module.js | 1.53 kB | 1.53 kB | ✓ |
| _app.module.js gzip | 745 B | 745 B | ✓ |
| _error.module.js | 10.3 kB | 10.3 kB | ✓ |
| _error.module.js gzip | 4.24 kB | 4.24 kB | ✓ |
| hooks.module.js | 651 B | 651 B | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js | 276 B | 276 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js | 5.44 kB | 5.44 kB | ✓ |
| link.module.js gzip | 2.45 kB | 2.45 kB | ✓ |
| routerDirect.module.js | 386 B | 386 B | ✓ |
| routerDirect..dule.js gzip | 275 B | 275 B | ✓ |
| withRouter.module.js | 396 B | 396 B | ✓ |
| withRouter.m..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 19 kB | 19 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _buildManifest.js | 244 B | 244 B | ✓ |
| _buildManifest.js gzip | 200 B | 200 B | ✓ |
| _buildManifest.module.js | 251 B | 251 B | ✓ |
| _buildManife..dule.js gzip | 210 B | 210 B | ✓ |
| Overall change | 495 B | 495 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _error.js | 256 kB | 256 kB | ✓ |
| _error.js gzip | 67.9 kB | 67.9 kB | ✓ |
| hooks.html | 4.19 kB | 4.19 kB | ✓ |
| hooks.html gzip | 1.07 kB | 1.07 kB | ✓ |
| index.js | 256 kB | 256 kB | ✓ |
| index.js gzip | 68 kB | 68 kB | ✓ |
| link.js | 288 kB | 288 kB | ✓ |
| link.js gzip | 77.1 kB | 77.1 kB | ✓ |
| routerDirect.js | 281 kB | 281 kB | ✓ |
| routerDirect.js gzip | 75.1 kB | 75.1 kB | ✓ |
| withRouter.js | 281 kB | 281 kB | ✓ |
| withRouter.js gzip | 75.2 kB | 75.2 kB | ✓ |
| Overall change | 1.37 MB | 1.37 MB | ✓ |

</details>

</details>


Commit: 91eaf1d77b14fb2d14f585b8a65ea715f7f0d917

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Server Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| buildDuration | 12.7s | 12.7s | ⚠️ +46ms |
| nodeModulesSize | 48.6 MB | 48.6 MB | ⚠️ +77 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| main-HASH.js | 18.1 kB | 18.1 kB | ✓ |
| main-HASH.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-HASH.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..4086094cc.js | 19.8 kB | 19.8 kB | ✓ |
| 16b1a7225520..94cc.js gzip | 7.37 kB | 7.37 kB | ✓ |
| 4952ddcd88e7..49d7903c2.js | 9.6 kB | 9.6 kB | ✓ |
| 4952ddcd88e7..03c2.js gzip | 3.76 kB | 3.76 kB | ✓ |
| commons.HASH.js | 10.4 kB | 10.4 kB | ✓ |
| commons.HASH.js gzip | 3.91 kB | 3.91 kB | ✓ |
| de003c3a9d30..64b97c6d1.js | 21.1 kB | 21.1 kB | ✓ |
| de003c3a9d30..c6d1.js gzip | 8.15 kB | 8.15 kB | ✓ |
| framework.HASH.js | 125 kB | 125 kB | ✓ |
| framework.HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 206 kB | 206 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| main-HASH.module.js | 14.1 kB | 14.1 kB | ✓ |
| main-HASH.module.js gzip | 5.33 kB | 5.33 kB | ✓ |
| webpack-HASH.module.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..95.module.js | 21.1 kB | 21.1 kB | ✓ |
| 16b1a7225520..dule.js gzip | 8.14 kB | 8.14 kB | ✓ |
| 4952ddcd88e7..cc.module.js | 11.9 kB | 11.9 kB | ✓ |
| 4952ddcd88e7..dule.js gzip | 4.41 kB | 4.41 kB | ✓ |
| de003c3a9d30..f3.module.js | 14.9 kB | 14.9 kB | ✓ |
| de003c3a9d30..dule.js gzip | 6.11 kB | 6.11 kB | ✓ |
| framework.HASH.module.js | 125 kB | 125 kB | ✓ |
| framework.HA..dule.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 189 kB | 189 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| polyfills-HASH.js | 15.3 kB | 15.3 kB | ✓ |
| polyfills-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| Overall change | 15.3 kB | 15.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _app.js | 2.92 kB | 2.92 kB | ✓ |
| _app.js gzip | 1.32 kB | 1.32 kB | ✓ |
| _error.js | 13.2 kB | 13.2 kB | ✓ |
| _error.js gzip | 5.1 kB | 5.1 kB | ✓ |
| hooks.js | 1.92 kB | 1.92 kB | ✓ |
| hooks.js gzip | 941 B | 941 B | ✓ |
| index.js | 318 B | 318 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js | 6.76 kB | 6.76 kB | ✓ |
| link.js gzip | 2.88 kB | 2.88 kB | ✓ |
| routerDirect.js | 413 B | 413 B | ✓ |
| routerDirect.js gzip | 285 B | 285 B | ✓ |
| withRouter.js | 423 B | 423 B | ✓ |
| withRouter.js gzip | 284 B | 284 B | ✓ |
| Overall change | 25.9 kB | 25.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _app.module.js | 1.53 kB | 1.53 kB | ✓ |
| _app.module.js gzip | 745 B | 745 B | ✓ |
| _error.module.js | 10.3 kB | 10.3 kB | ✓ |
| _error.module.js gzip | 4.24 kB | 4.24 kB | ✓ |
| hooks.module.js | 651 B | 651 B | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js | 276 B | 276 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js | 5.44 kB | 5.44 kB | ✓ |
| link.module.js gzip | 2.45 kB | 2.45 kB | ✓ |
| routerDirect.module.js | 386 B | 386 B | ✓ |
| routerDirect..dule.js gzip | 275 B | 275 B | ✓ |
| withRouter.module.js | 396 B | 396 B | ✓ |
| withRouter.m..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 19 kB | 19 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _buildManifest.js | 244 B | 244 B | ✓ |
| _buildManifest.js gzip | 200 B | 200 B | ✓ |
| _buildManifest.module.js | 251 B | 251 B | ✓ |
| _buildManife..dule.js gzip | 210 B | 210 B | ✓ |
| Overall change | 495 B | 495 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| index.html | 4.06 kB | 4.06 kB | ✓ |
| index.html gzip | 1.03 kB | 1.03 kB | ✓ |
| link.html | 4.58 kB | 4.58 kB | ✓ |
| link.html gzip | 1.11 kB | 1.11 kB | ✓ |
| withRouter.html | 4.59 kB | 4.59 kB | ✓ |
| withRouter.html gzip | 1.1 kB | 1.1 kB | ✓ |
| Overall change | 13.2 kB | 13.2 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Serverless Mode</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| buildDuration | 13.4s | 13.4s | ⚠️ +24ms |
| nodeModulesSize | 48.6 MB | 48.6 MB | ⚠️ +77 B |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| main-HASH.js | 18.1 kB | 18.1 kB | ✓ |
| main-HASH.js gzip | 6.39 kB | 6.39 kB | ✓ |
| webpack-HASH.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..4086094cc.js | 19.8 kB | 19.8 kB | ✓ |
| 16b1a7225520..94cc.js gzip | 7.37 kB | 7.37 kB | ✓ |
| 4952ddcd88e7..49d7903c2.js | 9.6 kB | 9.6 kB | ✓ |
| 4952ddcd88e7..03c2.js gzip | 3.76 kB | 3.76 kB | ✓ |
| commons.HASH.js | 10.4 kB | 10.4 kB | ✓ |
| commons.HASH.js gzip | 3.91 kB | 3.91 kB | ✓ |
| de003c3a9d30..64b97c6d1.js | 21.1 kB | 21.1 kB | ✓ |
| de003c3a9d30..c6d1.js gzip | 8.15 kB | 8.15 kB | ✓ |
| framework.HASH.js | 125 kB | 125 kB | ✓ |
| framework.HASH.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 206 kB | 206 kB | ✓ |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons) Modern</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| main-HASH.module.js | 14.1 kB | 14.1 kB | ✓ |
| main-HASH.module.js gzip | 5.33 kB | 5.33 kB | ✓ |
| webpack-HASH.module.js | 1.53 kB | 1.53 kB | ✓ |
| webpack-HASH..dule.js gzip | 746 B | 746 B | ✓ |
| 16b1a7225520..95.module.js | 21.1 kB | 21.1 kB | ✓ |
| 16b1a7225520..dule.js gzip | 8.14 kB | 8.14 kB | ✓ |
| 4952ddcd88e7..cc.module.js | 11.9 kB | 11.9 kB | ✓ |
| 4952ddcd88e7..dule.js gzip | 4.41 kB | 4.41 kB | ✓ |
| de003c3a9d30..f3.module.js | 14.9 kB | 14.9 kB | ✓ |
| de003c3a9d30..dule.js gzip | 6.11 kB | 6.11 kB | ✓ |
| framework.HASH.module.js | 125 kB | 125 kB | ✓ |
| framework.HA..dule.js gzip | 39.4 kB | 39.4 kB | ✓ |
| Overall change | 189 kB | 189 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| polyfills-HASH.js | 15.3 kB | 15.3 kB | ✓ |
| polyfills-HASH.js gzip | 4.75 kB | 4.75 kB | ✓ |
| Overall change | 15.3 kB | 15.3 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _app.js | 2.92 kB | 2.92 kB | ✓ |
| _app.js gzip | 1.32 kB | 1.32 kB | ✓ |
| _error.js | 13.2 kB | 13.2 kB | ✓ |
| _error.js gzip | 5.1 kB | 5.1 kB | ✓ |
| hooks.js | 1.92 kB | 1.92 kB | ✓ |
| hooks.js gzip | 941 B | 941 B | ✓ |
| index.js | 318 B | 318 B | ✓ |
| index.js gzip | 222 B | 222 B | ✓ |
| link.js | 6.76 kB | 6.76 kB | ✓ |
| link.js gzip | 2.88 kB | 2.88 kB | ✓ |
| routerDirect.js | 413 B | 413 B | ✓ |
| routerDirect.js gzip | 285 B | 285 B | ✓ |
| withRouter.js | 423 B | 423 B | ✓ |
| withRouter.js gzip | 284 B | 284 B | ✓ |
| Overall change | 25.9 kB | 25.9 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages Modern</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _app.module.js | 1.53 kB | 1.53 kB | ✓ |
| _app.module.js gzip | 745 B | 745 B | ✓ |
| _error.module.js | 10.3 kB | 10.3 kB | ✓ |
| _error.module.js gzip | 4.24 kB | 4.24 kB | ✓ |
| hooks.module.js | 651 B | 651 B | ✓ |
| hooks.module.js gzip | 371 B | 371 B | ✓ |
| index.module.js | 276 B | 276 B | ✓ |
| index.module.js gzip | 212 B | 212 B | ✓ |
| link.module.js | 5.44 kB | 5.44 kB | ✓ |
| link.module.js gzip | 2.45 kB | 2.45 kB | ✓ |
| routerDirect.module.js | 386 B | 386 B | ✓ |
| routerDirect..dule.js gzip | 275 B | 275 B | ✓ |
| withRouter.module.js | 396 B | 396 B | ✓ |
| withRouter.m..dule.js gzip | 274 B | 274 B | ✓ |
| Overall change | 19 kB | 19 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _buildManifest.js | 244 B | 244 B | ✓ |
| _buildManifest.js gzip | 200 B | 200 B | ✓ |
| _buildManifest.module.js | 251 B | 251 B | ✓ |
| _buildManife..dule.js gzip | 210 B | 210 B | ✓ |
| Overall change | 495 B | 495 B | ✓ |

</details>

<details>
<summary><strong>Serverless bundles</strong></summary>

|  | zeit/next.js canary | merceyz/next.js preset-modules | Change |
| - | - | - | - |
| _error.js | 256 kB | 256 kB | ✓ |
| _error.js gzip | 67.9 kB | 67.9 kB | ✓ |
| hooks.html | 4.19 kB | 4.19 kB | ✓ |
| hooks.html gzip | 1.07 kB | 1.07 kB | ✓ |
| index.js | 256 kB | 256 kB | ✓ |
| index.js gzip | 68 kB | 68 kB | ✓ |
| link.js | 288 kB | 288 kB | ✓ |
| link.js gzip | 77.1 kB | 77.1 kB | ✓ |
| routerDirect.js | 281 kB | 281 kB | ✓ |
| routerDirect.js gzip | 75.1 kB | 75.1 kB | ✓ |
| withRouter.js | 281 kB | 281 kB | ✓ |
| withRouter.js gzip | 75.2 kB | 75.2 kB | ✓ |
| Overall change | 1.37 MB | 1.37 MB | ✓ |

</details>

</details>


Commit: b2e0dafe9ce61baebb46f230dbff2266c0a3297b

**Comment by developit** [comment]
Just a note - the upstream PR has been merged and is released as [@babel/preset-modules@0.1.2](https://github.com/babel/preset-modules/releases/tag/0.1.2).

**Pull request merged**